lib/routes/routeBackbeat.js

```diff
@@ -1289,7 +1289,12 @@ function routeBackbeat(clientIP, request, response, log) {
                     [request.query.operation](request, response, log, next);
             }
             const versioningConfig = bucketInfo.getVersioningConfiguration();
-            if (!versioningConfig || versioningConfig.Status !== 'Enabled') {
+            // The following makes sure that only replication destination-related operations
+            // target buckets with versioning enabled.
+            // This allows lifecycle expiration operations (getMetadata) to work on non-versioned buckets.
+            const isCRRDestinationRequest = request.method === 'PUT' &&
+                (request.resourceType === 'data' || request.resourceType === 'metadata');
+            if (isCRRDestinationRequest && (!versioningConfig || versioningConfig.Status !== 'Enabled')) {
```

Comment:
I'd rather make this generic and let the caller (i.e. backbeat, who has full context!) specify if versioning is required, without introducing replication-related logic (which could actually change!) here.

something like:

```suggestion
            if (request.headers['x-scal-if-versionning-enabled'] && (!versioningConfig || versioningConfig.Status !== 'Enabled')) {
```

Reply:
updated.